python/sdk/test/integration_test.py

```diff
@@ -755,3 +755,55 @@ def test_custom_model_without_artifact(integration_test_url, project_name, use_g
 
     # Undeploy other running model version endpoints
     undeploy_all_version()
+
+@pytest.mark.integration
+def test_custom_model_with_artifact(integration_test_url, project_name, use_google_oauth):
+    merlin.set_url(integration_test_url, use_google_oauth=use_google_oauth)
+    merlin.set_project(project_name)
+    merlin.set_model("custom-w-artifact", ModelType.CUSTOM)
+    undeploy_all_version()
+
+    resource_request = ResourceRequest(1, 1, "1", "1Gi")
+    model_dir = "test/custom-model"
+    BST_FILE = "model.bst"
+
+    with merlin.new_model_version() as v:
+        iris = load_iris()
+        y = iris['target']
+        X = iris['data']
+        dtrain = xgb.DMatrix(X, label=y)
+        param = {'max_depth': 6,
+                 'eta': 0.1,
+                 'silent': 1,
+                 'nthread': 4,
+                 'num_class': 10,
+                 'objective': 'multi:softmax'
+                 }
+        xgb_model = xgb.train(params=param, dtrain=dtrain)
+        model_file = os.path.join((model_dir), BST_FILE)
+        xgb_model.save_model(model_file)
```

Comment:
nit: I think it's better to move the training code outside of `new_model_version()` scope for better readability and clarity. It would be like this:

```
# Train model
model_dir = "test/custom-model"
BST_FILE = "model.bst"

iris = load_iris()
y = iris['target']
X = iris['data']
dtrain = xgb.DMatrix(X, label=y)
param = {'max_depth': 6,
         'eta': 0.1,
         'silent': 1,
         'nthread': 4,
         'num_class': 10,
         'objective': 'multi:softmax'
         }
xgb_model = xgb.train(params=param, dtrain=dtrain)
model_file = os.path.join((model_dir), BST_FILE)
xgb_model.save_model(model_file)

# Deploy model
with merlin.new_model_version() as v:
         v.log_custom_model(image="ghcr.io/tiopramayudi/custom-predictor-go:v0.2", model_dir=model_dir)
```